presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -8512,6 +8512,34 @@ public void testComparisonWithLike()
                 "ON t1.orderkey=t2.orderkey");
     }
 
+    @Test
+    public void testSetUnion()
+    {
+        // sanity
+        assertQuery(
+                "select set_union(x) from (values array[1, 2], array[3, 4], array[5, 6]) as t(x)",
+                "select array[1, 2, 3, 4, 5, 6]");
+        assertQuery(
+                "select set_union(x) from (values array[1, 2, 3], array[2, 3, 4], array[7, 8]) as t(x)",
+                "select array[1, 2, 3, 4, 7, 8]");
+        // all nulls should return empty array to match behavior of array_distinct(flatten(array_agg(x)))
+        assertQuery(
+                "select set_union(x) from (values null, null, null) as t(x)",
+                "select array[]");
+        // nulls inside arrays should be captured while pure nulls should be ignored
+        assertQuery(
+                "select set_union(x) from (values null, array[null], null) as t(x)",
+                "select array[null]");
+        // null inside arrays should be captured
+        assertQuery(
+                "select set_union(x) from (values array[1, 2, 3], array[null], null) as t(x)",
+                "select array[1, 2, 3, null]");
+        // return null for empty rows
+        assertQuery(
```

Comment:
Can we add some tests with `group by`s  as well?

Reply:
Done